frame/include/bli_extern_defs.h

```diff
@@ -35,16 +35,26 @@
 #ifndef BLIS_EXTERN_DEFS_H
 #define BLIS_EXTERN_DEFS_H
 
-extern obj_t BLIS_TWO;
-extern obj_t BLIS_ONE;
-//extern obj_t BLIS_ONE_HALF;
-extern obj_t BLIS_ZERO;
-//extern obj_t BLIS_MINUS_ONE_HALF;
-extern obj_t BLIS_MINUS_ONE;
-extern obj_t BLIS_MINUS_TWO;
-
-extern thrcomm_t BLIS_SINGLE_COMM;
-extern thrinfo_t BLIS_PACKM_SINGLE_THREADED;
-extern thrinfo_t BLIS_GEMM_SINGLE_THREADED;
+#if !defined(BLIS_ENABLE_SHARED) || !defined(_MSC_VER)
+#define BLIS_EXPORT
+#else
+#ifdef BLIS_IS_BUILDING_LIBRARY
+#define BLIS_EXPORT __declspec(dllexport)
+#else
+#define BLIS_EXPORT __declspec(dllimport)
+#endif
+#endif
+
+BLIS_EXPORT extern obj_t BLIS_TWO;
+BLIS_EXPORT extern obj_t BLIS_ONE;
+//BLIS_EXPORT extern obj_t BLIS_ONE_HALF;
+BLIS_EXPORT extern obj_t BLIS_ZERO;
+//BLIS_EXPORT extern obj_t BLIS_MINUS_ONE_HALF;
+BLIS_EXPORT extern obj_t BLIS_MINUS_ONE;
+BLIS_EXPORT extern obj_t BLIS_MINUS_TWO;
+
+BLIS_EXPORT extern thrcomm_t BLIS_SINGLE_COMM;
+BLIS_EXPORT extern thrinfo_t BLIS_PACKM_SINGLE_THREADED;
+BLIS_EXPORT extern thrinfo_t BLIS_GEMM_SINGLE_THREADED;
```

Comment:
@fgvanzee, are these the only global data variables used in blis?

Reply:
Are you asking about:
1. *any* global variable (including those for internal use only), or
2. global variables that need to be exported as part of the public API

If (1), there are others, for example: `BLIS_ONE`, `BLIS_ZERO`, etc. in `frame/base/bli_const.c`. If (2), there are many others, though virtually all of them are declared `static` and only accessed via functions defined from within the same file.

Reply:
I was asking about part 2.

Reply:
Okay, I understand now. BTW, I got my explanation backwards. Sorry for the confusion--it's late and I'm normally getting ready for bed now. :)

Let me rephrase. If you are only interested in variables that would need to get exported for the DLL, then you definitely need to include the `BLIS_ONE`, `BLIS_ZERO`, variables in `frame/base/bli_const.c`. In fact, I would say that those are the *only* global variables that belong to the "public" API. The ones above, such as `BLIS_SINGLE_COMM`, are mostly for internal use and do not need to be exported.